src/main.rs

```diff
@@ -2,261 +2,60 @@
-                );
-            }
-            warp::reply::with_status("No login needed".to_string(), warp::http::StatusCode::OK)
-        })
-        .with(warp::reply::with::header(
-            "Access-Control-Allow-Origin",
-            "*",
-        ));
-
-    // The spans for these are covered in the handlers
-    let terminal_route = warp::path("ws")
-        .and(warp::path("term"))
-        .and(warp::ws())
-        .map(|ws: warp::ws::Ws| ws.on_upgrade(socket_handlers::term_handler));
-
-    let socket_route = warp::path("ws")
-        .and(warp::ws())
-        .map(|ws: warp::ws::Ws| ws.on_upgrade(socket_handlers::socket_handler));
-
-    let file_route = warp::path("ws")
-        .and(warp::path("file"))
-        .and(warp::ws())
-        .map(|ws: warp::ws::Ws| ws.on_upgrade(socket_handlers::file_handler));
-
-    #[cfg(feature = "frontend")]
-    let main_route = warp::any()
-        .map(|| {
-            let _guard = tracing::info_span!("main_route").entered();
-            let file = handle_error!(
-                DIR.get_file("index.html")
-                    .context("Couldn't get main HTML file"),
-                return warp::reply::with_status(
-                    "Couldn't get main HTML file",
-                    warp::http::StatusCode::INTERNAL_SERVER_ERROR
-                )
-                .into_response()
-            )
-            .contents();
-            warp::reply::html(file).into_response()
-        })
-        .with(warp::reply::with::headers(headers));
-
-    #[cfg(feature = "frontend")]
-    let page_routes = favicon_route.or(assets_route).or(main_route);
-
-    let socket_routes = terminal_route.or(file_route).or(socket_route);
-
-    let routes = socket_routes.or(login_route);
-    #[cfg(feature = "frontend")]
-    let routes = routes.or(page_routes);
-    let routes = routes.with(warp::trace::trace(|info| {
-        let remote_addr = info
-            .remote_addr()
-            .unwrap_or_else(|| std::net::SocketAddr::from((std::net::Ipv4Addr::UNSPECIFIED, 0)))
-            .ip();
-        let span = tracing::info_span!("request", %remote_addr);
-        span.in_scope(|| {
-            tracing::info!("Request to {}", info.path());
-            tracing::debug!(
-                "by {}, using {} {:?}",
-                info.user_agent().unwrap_or("unknown"),
-                remote_addr,
-                info.version(),
-            );
-        });
-        span
-    }));
-
-    let addr = IpAddr::from([0; 8]);
+    let addr = std::net::SocketAddr::from((IpAddr::from([0; 8]), CONFIG.port));
 
     if CONFIG.tls {
-        warp::serve(routes)
-            .tls()
-            .cert_path(&CONFIG.cert)
-            .key_path(&CONFIG.key)
-            .run((addr, CONFIG.port))
-            .await;
+        simple_hyper_server_tls::hyper_from_pem_files(
+            &CONFIG.cert,
+            &CONFIG.key,
+            simple_hyper_server_tls::Protocols::HTTP1,
```

Comment:
Any reason to not use:
```suggestion
            simple_hyper_server_tls::Protocols::ALL,
```
so clients can choose the HTTP protocol version?

Reply:
At the moment, to cut down on size, only HTTP/1.1 support is compiled in the binary (https://github.com/ravenclaw900/DietPi-Dashboard/blob/hyper/Cargo.toml#L27, https://github.com/ravenclaw900/DietPi-Dashboard/blob/hyper/Cargo.toml#L29)

Reply:
Okay, actually makes sense as long as we do not make explicit use of any HTTP/2 features.